### DISCUSSION TITLE
Feature(s) request

### DISCUSSION THREAD

**Discussion by Collins1975** [solectrus/solectrus#1030] [Ideas]
Hi Georg,

thank you so much for creating such a great tool for SENEC customers such as myself. Solectrus is incomparably a much nicer tool than the SENEC App and website and gives everyone a great (practically) live view of the photovoltaic, both current and also historical data. Great job!
This said, I would like to check and request if it is possible to add a feature or two to the tool?
Please bear in mind that I am making these comments strictly from an end user point of view. I am not a programmer myself and have absolutely no intention - and also not the necessary knowledge - to perform any changes in the provided tools and codes. I only mention this, in case I am missing something which is obvious for everyone else and I just need to "go here or there" to perform the necessary changes...

I installed the tool at my house and at a friends house. One thing that always irritates me is that they are all called the same: Solectrus. 

1. Would be good if the owners were able to include some individual information on the UI (e.g. "Collins's PV")
2. An option to display some details of the personal PV set-up, while keeping it anonymous (e.g. qty. and wattage of the panels, FORECAST_KWP, FORECAST_DECLINATION, FORECAST_AZIMUTH, maybe state (Bundesland)). I believe this could be quite useful to check how your PV is performing compared to others
3. Possibly add a kWP per Module view so users who live close to each other can check if their PV is performing similarly compared to their neighbours/friends PV's? 
4. My house has an air-heating pump and an activated cascade solution in my PV. This means that I am using my PV, both for powering my house and also for my heating and warm water. Is there a way to display these values separately, meaning what is used from the heating and what is used by the house power consumption?
5. Due to the latest power cost increases, I started a SENEC.Cloud 3.0 contract with SENEC on the 1st of February 22. For this purpose I purchased a package of 2.500 kWh for home power. I also activated the heating option. Since SENEC themselves are currently not able to display weather you are in plus or minus (based on the package you purchased, vs. what you consume/d), it would be fantastic if Solectrus had a view which displays this information?
6. A possibility to forward the information from Solectrus per email perhaps (e.g. in a fritzbox kind of way, once per week?)

This particular request may be completely ridiculous go totally against your concept. This said I feel the need to ask anyway: create a Solectrus app? 

Once again thank you for your time and support on your great tool! I am fully aware that you only do this in your free time and I have no intention whatsoever to cause you any trouble with this. Just hoping the one or the other topic may also make sense to yourself or other users out there and that if you can find the spare time and this is possible, that one day we could make the tool even better!

Thanks again and have a great day!

Collins

**Comment by holiday-sunrise** [comment]
[Collins1975](https://github.com/Collins1975) >> This said I feel the need to ask anyway: create a Solectrus app?

I have created a WEBAPP link :-) maybe it helps

**Reply by Collins1975** [reply]
Sorry für die späte Antwort und danke für Deinen Vorschlag @holiday-sunrise. Ich habe das auch bei mir so gelöst. Ich meinte aber eher im Bezug auf App Updates und Ähnliches. Ich bin schon sehr zufrieden mit Solectrus, wie es ist, aber automatische Updates wären klasse! :-)
Viele Grüße

**Reply by holiday-sunrise** [reply]
Hi du kannst deine docker container automatisch updaten lassen und der Link ist ja nur eine vier auf die UI. Somit wird immer alles aktuell angezeigt. Beste Grüße

**Comment by ledermann** [comment]
@Collins1975 I must have overlooked the original post above, my apologies for that.

Feature requests are always welcome, even though I can't promise if and when I can implement them. The work on SOLECTRUS takes place in between my "real" projects. But since I receive so much positive feedback, it certainly motivates me to continue pushing SOLECTRUS forward.

- Regarding 1 and 2: Personalizing the SOLECTRUS installation (name, etc.) is an interesting idea. The same goes for specifying additional parameters of the PV system and comparing them with others. Indeed, I have something in the works that heads in this direction, but it's not quite ready to be shown yet. It will take a while longer.
- Regarding 3: This is planned, see https://github.com/solectrus/solectrus/issues/1715
- Regarding 4: I'm afraid the heat pump is attributed to household consumption. I'm not certain, though, as our own heat pump installation is still pending. As soon as that's done, I'll dive deeper into it. 
- Regarding 5: SOLECTRUS directly queries the storage, so it doesn't make API requests to SENEC or the like. Hence, this is rather unlikely at the current stage.
- Regarding 6: Such reporting is conceivable. However, I'm uncertain if this is useful, since one can always look directly into SOLECTRUS and call up the weekly view.
- Regarding a native app: There likely won't be one in the foreseeable future, since the web app behaves almost like a native app. Regarding updates: I highly recommend [WatchTower](https://containrrr.dev/watchtower/), which can keep all Docker containers automatically up-to-date. I only mention this briefly in the installation guide, I need to elaborate on it there.

Thank you so much for your very friendly feedback!
